scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -152,99 +159,63 @@ function Install-SecurityDaemon {
-                foreach ($version in $SupportedBuildsForWindowsContainers) {
-                    Write-HostRed "$version"
-                }
-
-                $false
-            }
-            else {
-                $true
-            }
+        if (Test-MobyAlreadyInstalled) {
+            Write-HostRed
+            Write-HostRed 'IoT Edge Moby Engine is already installed. To reinstall, run `Uninstall-SecurityDaemon` first.'
+            return
         }
     }
-    if ($preRequisitesMet) {
-        Write-HostGreen "The container host is on supported build version $currentWindowsBuild."
-        Set-ContainerOs
-    }
-    else {
-        Write-HostRed
-        Write-HostRed ('The prerequisites for installation of the IoT Edge Security daemon are not met. ' +
-            'Please fix all known issues before rerunning this script.')
+
+    if (-not (Test-AgentRegistryArgs)) {
         return
     }
 
-    if (-not (Test-IotCore)) {
-        # `Invoke-WebRequest` may not use TLS 1.2 by default, depending on the specific release of Windows 10.
-        # This will be a problem if the release is downloaded from github.com since it only provides TLS 1.2.
-        # So enable TLS 1.2 in `[System.Net.ServicePointManager]::SecurityProtocol`, which enables it (in the current PS session)
-        # for `Invoke-WebRequest` and everything else that uses `System.Net.HttpWebRequest`
-        #
-        # This is not needed on IoT Core since its `Invoke-WebRequest` supports TLS 1.2 by default. It *can't* be done
-        # for IoT Core anyway because the `System.Net.ServicePointManager` type doesn't exist in its version of dotnet.
-        [System.Net.ServicePointManager]::SecurityProtocol =
-            [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::Tls12
+    if (-not (Setup-Environment $ContainerOs)) {
+        return
     }
 
-    if ($ExistingConfig -and (-not (Test-Path "$EdgeInstallDirectory\config.yaml"))) {
+    $configPath = Join-Path -Path $EdgeDataDirectory -ChildPath "config.yaml"
+    if ($ExistingConfig -and (-not (Test-Path $configPath))) {
         Write-HostRed
-        Write-HostRed "$EdgeInstallDirectory\config.yaml was not found."
+        Write-HostRed "$configPath was not found."
         return
     }
 
-    if ((-not $ExistingConfig) -and (Test-Path "$EdgeInstallDirectory\config.yaml")) {
+    if ((-not $ExistingConfig) -and (Test-Path $configPath)) {
         Write-HostRed
-        Write-HostRed "$EdgeInstallDirectory\config.yaml already exists."
+        Write-HostRed "$configPath already exists."
         Write-HostRed (
             'Delete it using `Uninstall-SecurityDaemon -Force -DeleteConfig` and then re-run `Install-SecurityDaemon`, ' +
             'or run `Install-SecurityDaemon -ExistingConfig` to use the existing config.yaml')
         return
     }
 
-    # Download
-    Get-SecurityDaemon
+    $restartNeeded = $false
+    if (-not $Finalize) {
```

Comment:
I am not sure I am following. This if around Get-SecurityDaemon is skipping it only if -Finalize is passed in, as it is already installed.

Interesting thought on the consistency, I do like that. So how about this: Deploy/Update-SD would be used anytime there is an update, Initialize-SD only needs to be used once. We could still keep Install, which would do Deploy and Initialize in one step.

Reply:
My bad. I missed the `-not`.

Agreed on having `Deploy-` and `Initialize-`, but I still think we should not have `Install-` at all. It seems it would be confusing to have a command named `Install-` but remind IoT Core users to not use it.

Reply:
Fixed.

Reply:
I think there is value in having a `Install-` wrapper cmdlet. It makes the instructions more streamlined for the 90% case . Install and unInstall has desirable congruity to usage.